opm/material/common/UniformTabulated2DFunction.hpp

```diff
@@ -186,21 +192,29 @@ class UniformTabulated2DFunction
      * \brief Evaluate the function at a given (x,y) position.
      *
      * If this method is called for a value outside of the tabulated
-     * range, a \c Opm::NumericalIssue exception is thrown.
+     * range, a \c Opm::NumericalIssue exception is thrown if
+     * throwUntabulated is true.
      */
     template <class Evaluation>
     Evaluation eval(const Evaluation& x, const Evaluation& y) const
     {
```

Comment:
What about instead adding` bool extrapolate=false` like for the UnifromXTabulated2DFunction. That would make the behavior more similar and make it up to the call side to determine whether extrapolation is a wanted feature or not.

Reply:
That is a good idea. Please be aware that at least for UniformXTabulated2DFunction, the call site is in opm-material (e.g. opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp) where the value is actually hardcoded. Using the same approach we could limit the option of extrapolation to only Co2GasPvt. Should we go down this route?

If yes the static member would move to Co2GasPvt, set in the same as before in simulators, and that value would be used to call the eval function in UniformTabulated2DFunction.

Which of the other call sites should it be used? There are
- opm/material/eos/PengRobinson.hpp
- opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.hpp

Reply:
BrineCo2Pvt.hpp.